### ISSUE TITLE
Unable to build on Windows

### ISSUE THREAD

**Issue by pjrader1** [anholt/libepoxy#196]
I'm trying to build libepoxy on Windows, but I keep running into compilation errors, most having to do with 'undeclared identifier' and 'undefined'. I'm using meson v0.48.1, ninja v1.8.2, and I've made sure to checkout the latest commit of libepoxy (`a7439f9`).

I've tried building using ninja, VS2017, and VS2015, all give the same error.

This is the output when I run `meson`, which appears to be normal:
```
λ meson
The Meson build system
Version: 0.48.1
Source dir: C:\Users\ifell\Developer\Git\libepoxy
Build dir: C:\Users\ifell\Developer\Git\libepoxy\build
Build type: native build
Project name: libepoxy
Project version: 1.5.4
Native C compiler: cl (msvc 19.15.26732.1)
Build machine cpu family: x86_64
Build machine cpu: x86_64
Has header "KHR/khrplatform.h": NO
Compiler for C supports arguments -we4002: YES
Compiler for C supports arguments -we4003: YES
Compiler for C supports arguments -w14010: YES
Compiler for C supports arguments -we4013: YES
Compiler for C supports arguments -w14016: YES
Compiler for C supports arguments -we4020: YES
Compiler for C supports arguments -we4021: YES
Compiler for C supports arguments -we4027: YES
Compiler for C supports arguments -we4029: YES
Compiler for C supports arguments -we4033: YES
Compiler for C supports arguments -we4035: YES
Compiler for C supports arguments -we4045: YES
Compiler for C supports arguments -we4047: YES
Compiler for C supports arguments -we4049: YES
Compiler for C supports arguments -we4053: YES
Compiler for C supports arguments -we4071: YES
Compiler for C supports arguments -we4819: YES
Library dl found: NO
Found Pkg-config: NO
Dependency gl found: YES
Dependency egl found: NO (tried pkgconfig)
Dependency x11 found: NO (tried pkgconfig)
Dependency glesv2 found: NO (tried pkgconfig)
Library libGLESv2 found: NO
Dependency glesv1_cm found: NO (tried pkgconfig)
Library libGLESv1_CM found: NO
Library opengl32 found: YES
Library gdi32 found: YES
Configuring config.h using configuration
Library dl found: NO
Checking for function "dlvsym" : NO
Build targets in project: 13
Found ninja-1.8.2 at 'C:\Program Files\Meson\ninja.EXE'
```

[Here's the output after running ninja.](https://pastebin.com/X9b7kNjU)

**Comment by pjrader1** [comment]
I set up my own AppVeyor instance of libepoxy, it built correctly (I tried it with both VS2015 and VS2017, both succeeded). For some reason on both my Windows machines, I get that error.

I did some investigating with the AppVeyor artifacts, I've discovered that `gl_generated.h` and `wgl_generated.h` are not the same on my machine compared to the AppVeyor build. Here's the missing lines from [`gl_generated.h`](https://pastebin.com/h5v5FqVu) and [`wgl_generated.h`](https://pastebin.com/Zby8nrS9). I'm not sure why my generated headers are different from the AppVeyor ones. If I manually replace the AppVeyor headers into my local build and run Visual Studio (skipping the header generation) I get further in the build. I don't know how libepoxy generates the files, so maybe this is intended behavior).

**Comment by pjrader1** [comment]
I figured out the problem. Meson, for some reason, was using Python 3.6.5, when I have Python 3.7 installed. When I ran `meson "runpython" <gen_dispatch.py>`, it didn't generate the whole file, however when I ran `python <gen_dispatch.py>` (Python 3) it generated correctly. This stumped me for about an hour until I decided to re-install Meson using `pip` (I had used Meson's .msi installer previously). Meson now uses Python 3.7, and `ninja` builds libepoxy correctly.
